moveit_ros/planning/planning_scene_monitor/test/trajectory_monitor_tests.cpp

```diff
@@ -0,0 +1,111 @@
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Abishalini Sivaraman */
+
+#include <memory>
+#include <chrono>
+#include <thread>
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "moveit/planning_scene_monitor/trajectory_monitor.h"
+#include "moveit/planning_scene_monitor/current_state_monitor.h"
+#include "moveit/utils/robot_model_test_utils.h"
+#include "rclcpp/node_interfaces/node_clock_interface.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tf2_ros/buffer.h"
+
+using namespace std::chrono_literals;
+
+struct MockTrajectoryMonitorMiddlewareHandle : public planning_scene_monitor::TrajectoryMonitor::MiddlewareHandle
+{
+  MOCK_METHOD(void, sleep, (), (override));
+};
+
+struct MockCurrentStateMonitorMiddlewareHandle : public planning_scene_monitor::CurrentStateMonitor::MiddlewareHandle
+{
+  MOCK_METHOD(rclcpp::Time, now, (), (const, override));
+  MOCK_METHOD(void, createJointStateSubscription,
+              (const std::string& topic, planning_scene_monitor::JointStateUpdateCallback callback), (override));
+  MOCK_METHOD(void, resetJointStateSubscription, (), (override));
+  MOCK_METHOD(std::string, getJointStateTopicName, (), (const, override));
+};
+
+void waitFor(std::chrono::seconds timeout, std::function<bool()> done)
+{
+  const auto start = std::chrono::steady_clock::now();
+  while ((std::chrono::steady_clock::now() - start) < timeout)
+  {
+    if (done())
+    {
+      return;
+    }
+    std::this_thread::sleep_for(1ms);
+  }
+}
+
+TEST(TrajectoryMonitorTests, SleepAtLeastOnce)
+{
+  auto mock_trajectory_monitor_middleware_handle = std::make_unique<MockTrajectoryMonitorMiddlewareHandle>();
+  auto mock_current_state_monitor_middleware_handle = std::make_unique<MockCurrentStateMonitorMiddlewareHandle>();
+
+  // THEN we expect it to call sleep function
+  EXPECT_CALL(*mock_trajectory_monitor_middleware_handle, sleep).Times(::testing::AtLeast(1));
+
+  // GIVEN a TrajectoryMonitor is started
+  planning_scene_monitor::CurrentStateMonitorPtr current_state_monitor;
+
+  current_state_monitor = std::make_unique<planning_scene_monitor::CurrentStateMonitor>(
```

Comment:
Why is the declaration and instantiation on different lines? (this sort of practice opens you up to problems).

~Also, I'll bet you one keyboard that `CurrentStateMonitorPtr` is `std::shared_ptr<CurrentStateMonitor>` which would mean you're converting a unique to a shared??? Can you just use `auto` or `std::unique_ptr<CurrentStateMonitor>` because even if `CurrentStateMonitorPtr` is a `unique_ptr` the ambiguity is frustrating since the different pointers are used for different ownership lifetimes.~ 

https://stackoverflow.com/questions/37884728/does-c11-unique-ptr-and-shared-ptr-able-to-convert-to-each-others-type

Can you change this to `make_shared`?